### DISCUSSION TITLE
ActiveRecord::StatementInvalid: Mysql2::Error::ConnectionError

### DISCUSSION THREAD

**Discussion by medito** [postalserver/postal#1842] [Help with using Postal]
Hi all,

I have a bug on connection with MySQL instance, I must restart postal each 1 hour to unlock the queue of my messages

Please find below the error log :
`[HPVSXM] ActiveRecord::StatementInvalid: Mysql2::Error::ConnectionError: Lost connection to MySQL server during query: SELECT  `statistics`.* FROM `statistics` ORDER BY `statistics`.`id` ASC LIMIT 1
smtp_1      | [11] [2022-01-31T09:44:29.199] ERROR -- : [HPVSXM] /usr/local/bundle/gems/mysql2-0.5.3/lib/mysql2/client.rb:131:in `_query'
smtp_1      | [11] [2022-01-31T09:44:29.199] ERROR -- : [HPVSXM] /usr/local/bundle/gems/mysql2-0.5.3/lib/mysql2/client.rb:131:in `block in query'
smtp_1      | [11] [2022-01-31T09:44:29.199] ERROR -- : [HPVSXM] /usr/local/bundle/gems/mysql2-0.5.3/lib/mysql2/client.rb:130:in `handle_interrupt'
smtp_1      | [11] [2022-01-31T09:44:29.199] ERROR -- : [HPVSXM] /usr/local/bundle/gems/mysql2-0.5.3/lib/mysql2/client.rb:130:in `query'
smtp_1      | [11] [2022-01-31T09:44:29.199] ERROR -- : [HPVSXM] /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:187:in `block (2 levels) in execute'
smtp_1      | [11] [2022-01-31T09:44:29.199] ERROR -- : [HPVSXM] /usr/local/bundle/gems/activesupport-5.2.6/lib/active_support/dependencies/interlock.rb:48:in `block in permit_concurrent_loads'
smtp_1      | [11] [2022-01-31T09:44:29.199] ERROR -- : [HPVSXM] /usr/local/bundle/gems/activesupport-5.2.6/lib/a`

## Environment details 

 - OS: Debian 11
 - One VM for Postal instance
 - One VM for Mysql instance

PS :  The Postal instance and MariDB are separated on two VMs

Thanks a lot for you help.
Kind regards,
Mehdi

**Comment by willpower232** [comment]
> Lost connection to MySQL server

presuming that your MySQL VM has stayed up then I think you will need to increase some of the limits in the configuration, you should be able to search for them in github.

**Reply by medito** [reply]
Hi @willpower232,

I'm using these config, but I have the same error :/
```
[mysqld]
max_connections=1000
wait-timeout = 31536000
max_allowed_packet = 256M
innodb_buffer_pool_size = 1024M
innodb_flush_log_at_trx_commit = 2
net_write_timeout=3600
net_read_timeout=3600
```

**Reply by willpower232** [reply]
You could also have a look at #1812 which sounds similar, if all the settings are generous then you could try watching the memory on the server in case there isn't enough or something along those lines.

**Reply by medito** [reply]
@willpower232 
Always the some problem :/

**Reply by medito** [reply]
Hi All 
Please any updates ?
Thanks

**Reply by willpower232** [reply]
Have you checked the available memory on the server or read that issue for other ideas?
